great_expectations/checkpoint/types/checkpoint_result.py

```diff
@@ -288,14 +289,63 @@ def to_json_dict(self) -> dict:
         due to the location of the "great_expectations/types/__init__.py" and "great_expectations/core/util.py" modules
         make this refactoring infeasible at the present time.
         """
-        dict_obj: dict = self.to_raw_dict()
-        serializeable_dict: dict = convert_to_json_serializable(data=dict_obj)
+        serializeable_dict: dict = {
+            "run_id": self.run_id.to_json_dict(),
+            "run_results": convert_to_json_serializable(
+                data=recursively_convert_to_json_serializable(test_obj=self.run_results)
+            ),
+            "checkpoint_config": self.checkpoint_config.to_json_dict(),
+            "success": convert_to_json_serializable(data=self.success),
+        }
+        serializeable_dict = recursively_convert_to_json_serializable(
+            test_obj=serializeable_dict
+        )
         return serializeable_dict
 
+    def __getstate__(self):
+        """
+        In order for object to be picklable, its "__dict__" or or result of calling "__getstate__()" must be picklable.
+        """
+        return self.to_json_dict()
+
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        result = cls.__new__(cls)
+
+        memo[id(self)] = result
+
+        attributes_to_copy = set(CheckpointResultSchema().fields.keys())
+        for key in attributes_to_copy:
+            try:
+                value = self[key]
+                value_copy = safe_deep_copy(data=value, memo=memo)
+                setattr(result, key, value_copy)
+            except AttributeError:
+                pass
```

Comment:
Let's please log the error under `debug`

Reply:
@cdkini This breaks all tests that insist on "no log messages expected" -- so, regrettably, have to leave it like this for now.